includes/class-newspack-popups-model.php

```diff
@@ -853,6 +859,11 @@ class="<?php echo esc_attr( implode( ' ', $classes ) ); ?>"
 	 * @return string The generated markup.
 	 */
 	public static function generate_popup( $popup ) {
+		// If previewing a single prompt, override saved settings with preview settings.
+		if ( Newspack_Popups::previewed_popup_id() ) {
+			$popup = self::retrieve_preview_popup( Newspack_Popups::previewed_popup_id() );
+		}
```

Comment:
Why is this needed? In a preview situation, the argument to this method is already the preview popup:

https://github.com/Automattic/newspack-popups/blob/c8189dc9c6376fa89509bdf98b75e12f60c6f27f/includes/class-newspack-popups-inserter.php#L52-L55

Reply:
This is needed because for regular inline prompts, they're inserted into post content with [this `the_content` hook callback](https://github.com/Automattic/newspack-popups/blob/c8189dc9c6376fa89509bdf98b75e12f60c6f27f/includes/class-newspack-popups-inserter.php#L147). In this callback, they're created as shortcodes, which means they're generated by [this method](https://github.com/Automattic/newspack-popups/blob/c8189dc9c6376fa89509bdf98b75e12f60c6f27f/includes/class-newspack-popups-inserter.php#L345), which doesn't use the `popups_for_post` method where the preview popup is generated.

Reply:
Ok, now I get it - this was not needed before (before #505) because in a single prompt preview situation _every_ shortcode was displaying the previewed prompt. This was [changed in #505](https://github.com/Automattic/newspack-popups/pull/505/files#diff-84a23fd0b9ee98b91687d95435cc335c32a0f5c6cb217816816fe330ae72c4f3L354-R355), but as we see here it did introduce a regression, which the code here in the PR fixes.